README.adoc

```diff
@@ -1,5 +1,256 @@
+
+* `condtionExpresison`: With the help of the ConditionExpression, we can exclude cases that we do not want to generate if the condition is false. This should return a Boolean Spring expression.
+
+* `templateContext`: Template context is used to put expression result to template variable. We can refer to the expression through the template context name. (Not used often because of the helpers).
+** Example:
++
+We can refer to self through the applicationAsVariable template context name.
++
+[source,yaml]
+----
+    templateContext:
+        - name: applicationAsVariable
+          expression: "#self"
+----
+
+
+* `copy`: It can be used to copy a binary file. In this case the template file used as binary, no templating is performed.
+
+* `permission`: It can be used to add execution permission to files. For example bash files.
+
+** We use the POSIX permission pattern. Use this pattern `rwxrwxrwx` (read,write,execute).
+    This contains 3 permission type in this order:
+*** owner
+*** group
+*** other
+
+** We add the global permission first if it is present and if the template have a local permission we use that.
+
+** Example
+
++
+[source,yaml]
+----
+  - name: judo.sh
+    pathExpression: "'judo.sh'"
+    templateName: judo.sh.hbs
+    permission: rwxrwx---
+    actorTypeBased: false
+----
+
+* `exclude`: This parameter can be used in an override to exclude the given template from a generation. With this parameter only the `name` is effective.
+
+=== GeneratorModel
+
+The `GeneratorModel` Java class loads the Handlebars files from the base directory and overrides them with the files from the override directory if it is necessary.
+The class contains a global permission parameter that adds a global permission to all of the Handlebars files.
+(The `GeneratorTemplate` Java class, contained within the GeneratorModel class, includes all of the YAML representations listed earlier. The GeneratorModel maps the YAML properties to the GeneratorTemplate.)
+
+=== Generator Maven Project
+
+https://github.com/BlackBeltTechnology/judo-meta-esm/tree/develop/generator-maven-plugin#readme[This] link contains the description of the generator maven plugin.
+
+=== Helper registration
+
+We don't register helpers manually. To register a java helper class use the `@TemplateHelper` annotation.
+
+IMPORTANT: The Java class methods must be static.
+
+If you inherit your helper class from the `StaticMethodValueResolver` class, you can refer to its methods using dot notation.
+
+The helpers are available in springEL too with # prefix.
+
+=== Self meaning in different situations
+
+The self mean different thinks in the springEL and Handlebars depend on the YAML properties:
+
+- If actorTypeBased is enabled, then #self refers to the actor.
+
+- If we have a `factoryExpression`, then the self is the current iterated element of collection.
+
+- If we don't have a `factoryExpression` and it is not actorTypeBased, then #self refers to the model.
```

Comment:
> If we don't have a `factoryExpression` and it is not actorTypeBased, then

If it's not in context of a `factoryExpression` and actorTypeBased is disabled, then self...

Reply:
[4d7240d](https://github.com/BlackBeltTechnology/judo-generator-commons/pull/29/commits/4d7240d5f36e8ab753b70553ed41ed3371f1e13e)